packages/application-components/src/components/modal-pages/internals/modal-page.styles.ts

```diff
@@ -27,65 +24,94 @@ export const stylesBySize = {
+export const ModalOverlay = styled.div<{
+  size: TModalPageSize;
+  zIndex?: number;
+}>`
   position: absolute;
-  z-index: ${typeof props.zIndex === 'number'
-    ? // Use `!important` to overwrite the default value assigned by the Stacking Layer System.
-      `${props.zIndex} !important`
-    : 'auto'};
   top: 0;
   left: 0;
   width: 100%;
   height: 100%;
   background-color: hsla(195, 35%, 20%, 0.05);
+  z-index: ${({ zIndex }) =>
+    // Use `!important` to overwrite the default value assigned by the Stacking Layer System.
+    typeof zIndex === 'number' ? `${zIndex} !important` : 'inherit'};
   opacity: 0;
-  transition: transform ${stylesBySize[props.size].transitionTime}ms ease;
-`;
 
-export const getAfterOpenContainerAnimation = (): SerializedStyles => css`
-  transform: translate3d(0, 0, 0) !important;
+  ${({ size }) => {
+    const transitionDuration = stylesBySize[size].transitionTime;
+    return css`
+      animation: ${overlayShow} ${transitionDuration}ms ease-out forwards;
+    `;
+  }}
 `;
 
-export const getAfterOpenOverlayAnimation = (): SerializedStyles => css`
-  opacity: 1 !important;
-`;
+const getContentShowAnimation = (size: TModalPageSize) =>
+  keyframes({
+    '0%': {
+      opacity: 0,
+      transform: `translate3d(${
+        size !== 'scale' ? stylesBySize[size].width : '30px'
+      }, 0, 0)`,
+    },
+    '100%': {
+      opacity: 1,
+      transform: `translate3d(0, 0, 0)`,
+    },
+  });
 
-export const getBeforeCloseContainerAnimation = (
-  props: StyleProps
-): SerializedStyles => css`
-  transform: translate3d(
-    ${props.size !== 'scale' ? stylesBySize[props.size].width : '30px'},
-    ,
-    0,
-    0
-  ) !important;
-`;
+const getContentHideAnimation = (size: TModalPageSize) =>
+  keyframes({
+    '0%': {
+      opacity: 1,
+      transform: `translate3d(0, 0, 0)`,
+    },
+    '100%': {
+      opacity: 0,
+      transform: `translate3d(${
+        size !== 'scale' ? stylesBySize[size].width : '30px'
+      }, 0, 0)`,
+    },
+  });
+
+export const ModalContent = styled(Content, {
+  shouldForwardProp: (prop) => prop !== 'size',
+})<{ size: TModalPageSize }>`
+  position: absolute;
+  top: 0;
+  right: 0;
+  height: 100%;
+  width: ${({ size }) =>
+    size !== 'scale'
+      ? // In case we're using a specific size, we want it to be used until there's no space left.
+        // In such scenario, we want the modal to be as wide as possible, but using the shared indentation width size.
+        `min(${stylesBySize[size].width}, calc(100% - ${PORTALS_CONTAINER_INDENTATION_SIZE})) !important`
+      : stylesBySize.scale.width};
+  display: flex;
+  flex-direction: column;
+  background-color: ${designTokens.colorSurface};
+  box-shadow: 0px 0px 40px 0px rgba(0, 0, 0, 0.1);
+  outline: 0;
+  z-index: inherit;
+
+  &[data-state='open'] {
+    animation: ${({ size }) => getContentShowAnimation(size)} 300ms ease
+      forwards;
+  }
```

Comment:
The `data-state` attribute is added to the `Dialog.Content` component rendered in the DOM. Animations are currently keyed off this attribute.